### PR TITLE
refactor(android): migrate VideoEventEmitter to Kotlin

### PR DESCRIPTION
**Pull request by YangJonghun**
## Summary
refactor(android): migrate VideoEventEmitter to Kotlin

### Motivation
- remove unimplemented events(onPlaybackStalled, onPlaybackResume) (iOS only)
- remove receiveAdErrorEvent (duplicate of receiveAdEvent)

## Test plan

### CODE REVIEW

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java [hunk 1]
```diff
@@ -323,12 +324,6 @@ private boolean isPlayingAd() {
         return player != null && player.isPlayingAd();
     }
 
-    @Override
-    public void setId(int id) {
```
Comment (freeboub) [inline android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:327]: If I remember well this was necessary to support multiple <Video in the same page :/ but maybe I am wrong ...

**Review by freeboub — APPROVED** [review]

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java (hunk 1 above)
Comment (YangJonghun) [reply android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java:327]: View id will be inject within ReactExoplayerVIewManager.addEventEmitters method from this PR
fyi) https://github.com/TheWidlarzGroup/react-native-video/pull/3962/files#diff-910ac265923b0ad5faf4bbeb2cb1742f2608233f22f952c323925141f1cd8d8fR279-R285

**Review by KrzysztofMoch — APPROVED** [review]

File: android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+    EVENT_VIDEO_TRACKS("onVideoTracks"),
+    EVENT_ON_RECEIVE_AD_EVENT("onReceiveAdEvent");
+
+    companion object {
+        fun toMap() =
+            mutableMapOf<String, Any>().apply {
+                EventTypes.entries.forEach { eventType ->
+                    put("top${eventType.eventName.removePrefix("on")}", mapOf("registrationName" to eventType.eventName))
+                }
+            }
```
Comment (KrzysztofMoch) [inline android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt:53]: Will this work also with "old architecture"? If not we should add "if" here to don't do it on "old arch"

Comment (YangJonghun) [reply android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt:53]: It works properly on old arch

Comment (KrzysztofMoch) [reply android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt:53]: Thanks, then I will merge after CI is done

**Pull request merged**